iam/snippets/roles_view_grantable.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START iam_view_grantable_roles]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"golang.org/x/oauth2/google"
+	iam "google.golang.org/api/iam/v1"
+)
+
+// viewGrantableRoles lists roles grantable on a resource.
+func viewGrantableRoles(w io.Writer, fullResourceName string) ([]*iam.Role, error) {
```

Comment:
Might be helpful to document what `fullResourceName` looks like.

Reply:
This is in the documentation where the snippet will live: https://cloud.google.com/iam/docs/viewing-grantable-roles